services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy

```diff
@@ -631,7 +629,9 @@ class FlowRulesSpec extends HealthCheckSpecification {
         flowHelper.addFlow(flow)
 
         and: "Reproduce situation when switches have missing rules by deleting flow rules from them"
+        def flowInfoFromDb = database.getFlow(flow.id)
         def involvedSwitches = pathHelper.getInvolvedSwitches(flow.id)*.dpId
+        def transitSwitchIds = (involvedSwitches.size() > 2) ? involvedSwitches[-1..-2] : []
```

Comment:
You pick neighboring switches on L624. Why do you expect transit switches here?

Reply:
fixed

Reply:
the why do you assume 'no transit switches' situations then? (also during `|| true` below)

Reply:
fixed

Reply:
also the algorithm of finding not neighbouring noviflow switches was fixed